### PR TITLE
Test cases for lms applications

### PR DESCRIPTION
**Pull request by Syed-Ali-Abbas-Zaidi**
[LP-XXXX](https://philanthropyu.atlassian.net/browse/LP-XXXX)

[Theme PR](https://github.com/OmnipreneurshipAcademy/adg-edx-theme/pull/###)

***

#### Description:
Test cases for a single method or class from lms/applications:

- helpers.py
- handlers.py
- forms.py
- models.py
- views.py

***

## Things to Cross-check:

- [ ] Technical Docs updated if needed?
- [ ] [Domain Model](https://drive.google.com/file/d/1iRSqxDQQMXl6JzLw9nZtq1yGshwMBLTn/view?usp=sharing) updated if needed?
- [ ] [Core Changes Wiki](https://github.com/OmnipreneurshipAcademy/edx-platform/wiki/Core-Changes) updated if needed?

### CODE REVIEW

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/3359/workflows/workflows_id/jobs/3359/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_training.py (100%)
-------------
Total:   104 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/3361/workflows/workflows_id/jobs/3361/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_training.py (100%)
-------------
Total:   106 lines
Missing: 0 lines
Coverage: 100%
-------------
```

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/3363/workflows/workflows_id/jobs/3363/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_training.py (100%)
-------------
Total:   106 lines
Missing: 0 lines
Coverage: 100%
-------------
```

File: openedx/adg/lms/applications/tests/test_training.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    assert validate_logo_size(mocked_file) is None
+
+
+@pytest.mark.django_db
+def test_update_business_line_group():
+    """
+    Tests that the user group gets created/updated whenever the business line gets created/updated.
+    """
+    business_line = BusinessLineFactory(title=TITLE_BUSINESS_LINE_1)
+    business_line.save()
```
Comment (asadali145) [inline openedx/adg/lms/applications/tests/test_training.py:47]: No need to call save explicitly.

File: openedx/adg/lms/applications/tests/test_training.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    'business_line, interest_in_business, submit_or_back_clicked',
+    [
+        ('business_line', TEST_INTEREST_IN_BUSINESS, 'submit'),
+        ('business_line', TEST_INTEREST_IN_BUSINESS, 'back'),
+        ('', TEST_INTEREST_IN_BUSINESS, 'back'),
+        ('business_line', '', 'back'),
+        ('', '', 'back'),
+    ]
+)
+@mock.patch('openedx.adg.lms.applications.forms.validate_word_limit')
```
Comment (asadali145) [inline openedx/adg/lms/applications/tests/test_training.py:68]: Lets user mocker.

Comment (asadali145) [reply openedx/adg/lms/applications/tests/test_training.py:68]: Same for all tests.

File: openedx/adg/lms/applications/tests/test_training.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+def test_valid_business_line_interest_form(
```
Comment (asadali145) [inline openedx/adg/lms/applications/tests/test_training.py:69]: We can merge valid and invalid using params.

File: openedx/adg/lms/applications/tests/test_training.py [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+        'submit_or_back_clicked': 'submit'
+    }
+
+    form = BusinessLineInterestForm(data=test_data)
+    assert not form.is_valid()
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('business_line_admin, expected_result', [(True, True), (False, False)])
+def test_business_line_model_is_user_business_line_admin(business_line_admin, expected_result):
```
Comment (asadali145) [inline openedx/adg/lms/applications/tests/test_training.py:115]: Lets just rename this to `test_ is_user_business_line_admin`

File: openedx/adg/lms/applications/tests/test_training.py [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+    Tests the business lines according to the type of the user.
+    """
+    business_lines = [BusinessLineFactory()]
+    BusinessLineFactory()
+
+    test_user = UserFactory(groups=business_lines)
+    business_lines_for_user = BusinessLine.business_lines_for_user(test_user)
+    assert len(business_lines_for_user) == 1
+
+    test_admin_user = UserFactory(is_superuser=True)
```
Comment (asadali145) [inline openedx/adg/lms/applications/tests/test_training.py:140]: For admin user you should add adg admins group to user groups.

File: openedx/adg/lms/applications/tests/test_training.py [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+    Tests that application introduction post request is only for the users who hve not started their application.
+    """
+    test_user = UserFactory()
+    if application_exist:
+        ApplicationHubFactory(user=test_user)
+
+    request = RequestFactory().post(reverse('application_introduction'))
+    request.user = test_user
+
+    response = ApplicationIntroductionView.as_view()(request)
```
Comment (asadali145) [inline openedx/adg/lms/applications/tests/test_training.py:210]: We can also test methods like get_context_data of a view to test that valid data is returned, you can check sample tests from code base.

**Review by asadali145 — COMMENTED** [review]
Awesome work! Here are a few suggestions/improvments.

**Comment by omnipreneurshipA** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/3373/workflows/workflows_id/jobs/3373/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/develop...HEAD, staged and unstaged changes
-------------
openedx/adg/lms/applications/tests/test_training.py (100%)
-------------
Total:   102 lines
Missing: 0 lines
Coverage: 100%
-------------
```

File: openedx/adg/lms/applications/tests/test_training.py [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+    """
+    Tests application introduction context data.
+    """
+    test_user = UserFactory()
+
+    request = RequestFactory().get(reverse('application_introduction'))
+    request.user = test_user
+
+    response = ApplicationIntroductionView.as_view()(request)
+    assert 'view' in response.context_data
```
Comment (asadali145) [inline openedx/adg/lms/applications/tests/test_training.py:168]: I guess you know what is the original purpose of these tests, We should verify that all the data is present in the context_data. For that we can verify all the keys and data as well using mock data. But for now we are okay with this one.

**Review by asadali145 — APPROVED** [review]

**Pull request closed without merging**